src/nexus/serial.cljc

```diff
@@ -0,0 +1,109 @@
+        (cond-> (assoc ctx :actions (rest actions))
+          (seq results) (update :results conjv results)
+          (seq errors) (update :errors conjv errors)))
+      ;;Effect not found
+      (-> ctx
+          (assoc :actions (rest actions))
+          (update :errors conjv {:phase :execute-effect
+                                 :effect-k effect-id
+                                 :err (ex-info "No such effect" {:available-effects (keys (:nexus/effects nexus-map))})})))))
+
+(defn expand-first [{:nexus/keys [interceptors] :as nexus-map} {:keys [state dispatch-data actions errors] :as ctx}]
+  (let [[first-action & remaining] actions
+        [action-id & _action-params :as action] first-action]
+    (if-let [handler (get-in nexus-map [:nexus/actions action-id])]
+      (let [action-interpolated (nexus/interpolate-1 nexus-map dispatch-data action)
+            action-interceptor {:id action-id
+                                :phase :expand-action
+                                :before-action (partial nexus/wrap-action-handler handler)}
+            {:keys [action actions errors]} (nexus/run-interceptors
+                                             (cond-> {:state state :action action-interpolated}
+                                               errors (assoc :errors errors))
+                                             (conjv interceptors action-interceptor)
+                                             [:before-action :after-action :action])]
+        (cond-> (assoc ctx :actions (intov actions remaining))
+          (seq errors) (assoc :errors errors)
+          ;;TODO: Do we want to prevent prefixing the returned actions if they're invalid?
+          (not (nexus/actions? actions)) (update-in [:errors] conjv {:action action
+                                                                     :phase :expand-action
+                                                                     :err (ex-info (str (first action) " should expand to a collection of actions")
+                                                                                   {:res actions})})))
+      (-> ctx
+          (assoc :actions remaining)
+          (update-in [:errors] conjv
+                     {:action action
+                      :phase :expand-action
+                      :err (ex-info (str (first action) " not found in handler map")
+                                    {:available-actions (keys (:nexus/actions nexus-map))})})))))
+
+(defn expand-lazily [nexus-map {:keys [actions] :as ctx}]
+  (let [actions* (filterv some? actions)
+        ctx* (assoc ctx :actions actions*)
+        [[next-id & _next-params] & _remaining] actions*]
+    (if (or (empty? actions*) (contains? (:nexus/effects nexus-map) next-id))
+      ctx*
+      (recur nexus-map (expand-first nexus-map ctx*)))))
+
+(defn dispatch-serial-base [nexus-map {:keys [system dispatch-data] :as ctx}]
```

Comment:
Perhaps `dispatch-serial-base`  could be `dispatch-serially` instead? It isn't clear to me what "base" is in this context.